internals/plan/plan.go

```diff
@@ -42,11 +68,68 @@ const (
 	defaultCheckThreshold = 3
 )
 
+// layerExtensions keeps a map of registered extensions.
+var layerExtensions = map[string]LayerSectionExtension{}
+
+// layerBuiltins represents all the built-in layer sections. This list is used
+// for identifying built-in fields in this package. It is unit tested to match
+// the YAML fields exposed in the Layer type, to catch inconsistencies.
+var layerBuiltins = []string{"summary", "description", "services", "checks", "log-targets"}
+
+// RegisterExtension adds a plan schema extension. All registrations must be
+// done before the plan library is used.
+func RegisterExtension(field string, ext LayerSectionExtension) {
+	if slices.Contains(layerBuiltins, field) {
+		panic(fmt.Sprintf("internal error: extension %q already used as built-in field", field))
+	}
+	if _, ok := layerExtensions[field]; ok {
+		panic(fmt.Sprintf("internal error: extension %q already registered", field))
+	}
+	layerExtensions[field] = ext
+}
+
+// UnregisterExtension removes a plan schema extension. This is only
+// intended for use by tests during cleanup.
+func UnregisterExtension(field string) {
+	delete(layerExtensions, field)
+}
+
 type Plan struct {
 	Layers     []*Layer              `yaml:"-"`
 	Services   map[string]*Service   `yaml:"services,omitempty"`
 	Checks     map[string]*Check     `yaml:"checks,omitempty"`
 	LogTargets map[string]*LogTarget `yaml:"log-targets,omitempty"`
+
+	Sections map[string]LayerSection `yaml:",inline"`
+}
+
+// Section retrieves a section from the plan. If Section is called
+// before the plan is loaded, or with an unregistered field, this method
+// will return nil.
+func (p *Plan) Section(field string) LayerSection {
+	return p.Sections[field]
```

Comment:
What's the purpose of this method when its implementation is public and that trivial?

Reply:
I removed this as rightfully it no longer really serves a purpose. Thanks you. Resolving.